app/controllers/api/discussions_controller.rb

```diff
@@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
   load_resource only: [:create, :update]
 
   def dashboard
-    instantiate_collection { |collection| filter_collection collection }
-    respond_with_discussions
+    instantiate_collection { |collection| collection_for_dashboard collection }
+    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
+  end
+
+  def inbox
+    instantiate_collection { |collection| collection_for_inbox collection }
+    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
```

Comment:
Nope, I think this is begging for trouble. This leakage thing is the reason
we get weird behaviour like when you first load recent threads, it gives
you some threads that are recent, and some threads from a while ago before
it updates, with chunks missing in between. It means that when you 'load
more' recent threads, you can't be sure that you're getting the next 50
records (because you may have some older threads which already exist),
which means after they're loaded, the scroll position leaves you
disoriented (ie, after load, you're not looking at the same threads on the
screen as you were before the load; the user gets totally lost because some
threads are pushed down and others aren't.. and it's like 'okay, why when I
load more did it give me a bunch of threads from July 10th when I was
looking at a bunch of threads from July 4th before?')

Basically I've tried it out the way you're suggesting and found it to be a
pretty bad time.

On Mon, Aug 3, 2015 at 2:03 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/discussions_controller.rb
> <https://github.com/loomio/loomio/pull/2369#discussion_r36067096>:
>
> > @@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
> >    load_resource only: [:create, :update]
> >
> >    def dashboard
> > -    instantiate_collection { |collection| filter_collection collection }
> > -    respond_with_discussions
> > +    instantiate_collection { |collection| collection_for_dashboard collection }
> > +    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
> > +  end
> > +
> > +  def inbox
> > +    instantiate_collection { |collection| collection_for_inbox collection }
> > +    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
>
> I still don't understand why you would get leakage issues... if a
> discussion matches the filter it should be ok to display, regardless of
> what query it came from.. and if that isn't the case we're going to have
> problems when discussions come from other places ie: the message channel...
>
> So I can understand marking discussions which don't have readers, as they
> can be problematic.. but I think we should design the filters to be able to
> use discussions from any query if they match.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r36067096>.
>